### DISCUSSION TITLE
Remediating BitLocker DMA Exception Errors with Microsoft Intune - odds+endpoints

### DISCUSSION THREAD

**Discussion by giscus** [ennnbeee/oddsandendpoints-web#4] [Comments]
# Remediating BitLocker DMA Exception Errors with Microsoft Intune

Using Microsoft Intune platform powershell scripts to allow for DMA exceptions for specific hardware manufacturers to enable silent BitLocker disk encryption.

https://memv.ennbee.uk/posts/bitlocker-dma-exceptions/

**Comment by Martony78** [comment]
This is not true, DMA exceptions do not prevent silent encryption configured in Intune. It prevents automatic encryption.
Just ask copilot:
« what is the difference between Bitlocker automatic encryption and Bitlocker silent encryption »
Certainly! Let me explain the difference between BitLocker automatic encryption and BitLocker silent encryption:

1. **Automatic Encryption**:
   - **When**: Automatic encryption occurs during the Windows out-of-the-box experience (OOBE) mode.
   - **Devices**: It applies to modern standby devices or those compliant with the Hardware Security Test Interface (HSTI).
   - **User Interaction**: During automatic encryption, the end user doesn't see any prompts or UI related to BitLocker.
   - **Purpose**: It ensures that encryption is enabled seamlessly without user intervention.
   - **Context**: Typically happens during initial device setup.
   - **Note**: This is not the same as silent encryption.

2. **Silent Encryption**:
   - **Configuration**: You can configure a policy for BitLocker to automatically and silently encrypt a device without presenting any UI to the end user.
   - **User Role**: Even when the user isn't a local Administrator on the device, silent encryption can proceed.
   - **Purpose**: It allows IT administrators to enforce encryption without disrupting the user experience.
   - **Use Case**: Useful for organizations that want to ensure data protection without user involvement.
   - **Management**: Administrators can manage BitLocker recovery keys after silent encryption.
   - **Note**: Silent enablement is different from automatic encryption during OOBE².

In summary, automatic encryption happens during initial setup, while silent encryption allows IT admins to enforce encryption without user interaction. Both serve the purpose of securing data using BitLocker. 😊🔒¹²

Source : conversation avec Copilot, 10/07/2024
(1) Troubleshooting BitLocker with the Intune encryption report. https://learn.microsoft.com/en-us/troubleshoot/mem/intune/device-protection/troubleshoot-bitlocker-admin-center.
(2) Encrypt Windows devices with BitLocker in Intune - Microsoft Intune .... https://learn.microsoft.com/en-us/mem/intune/protect/encrypt-devices.
(3) Setting the BitLocker encryption algorithm for Autopilot devices. https://learn.microsoft.com/en-us/autopilot/bitlocker.
(4) Difference between Device Encryption and BitLocker - The Windows Club. https://www.thewindowsclub.com/difference-between-device-encryption-and-bitlocker. »
« does silent encryption occurs even if there are dma exceptions? »
« Yes, silent encryption can still occur even if there are DMA (Direct Memory Access) exceptions. DMA exceptions typically refer to scenarios where certain hardware components (such as network adapters or storage controllers) have direct access to a system's memory without CPU intervention. Despite these exceptions, BitLocker's silent encryption process can proceed, ensuring data protection without disrupting the user experience. 😊🔒 »

**Reply by ennnbeee** [reply]
Thanks for the feedback, I've made it clearer in this post that the DMA exceptions are blocking Automatic encryption.

**Reply by Martony78** [reply]
I can’t see the changes to the post, have you commited them?
I honestly don’t want to bother you, it’s just your posts are so useful and trusted by the community that it can make customers complaining against manufacturers about an issue that has nothing to do with them. Indeed, DMA exceptions save lots of companies who want to encrypt their drives with a 256 bits algorithm, since it avoids the disk to be encrypted with a 128 bits algorithm before autopilot kicks in

**Reply by ennnbeee** [reply]
Ahh, I think the draft got stuck, I've sorted that now and although I'm still mentioning Silent Encryption in the opening line, the rest of the post references Automatic Encryption throughout.
